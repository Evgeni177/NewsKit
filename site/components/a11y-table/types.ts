import {TableProps} from '../table';

export interface A11yTableProps {
  name: string;
  description: string;
  table: TableProps;
}
