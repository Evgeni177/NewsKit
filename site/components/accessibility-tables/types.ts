export interface AccessibilityTablesProps {
  focusOrder?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: FocusOrderProps[];
    };
  };
  aria?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: AriaProps[];
    };
  };
  interaction?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: InteractionProps[];
    };
  };
}

export interface FocusOrderProps {
  order: string[];
  element: string | JSX.Element;
  role?: string;
}

export interface AriaProps {
  category: string;
  attribute: string;
  value: string;
  description: string;
}

export interface InteractionProps {
  command: string[];
  description: string;
}
