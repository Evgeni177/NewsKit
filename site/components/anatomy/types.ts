import {ImageProps} from '../../../src/image/types';

export interface AnatomyProps {
  media: ImageProps;
  list: Array<string>;
}
