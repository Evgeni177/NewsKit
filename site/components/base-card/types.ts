import {CardProps} from 'newskit';

export interface BaseCardProps extends CardProps {
  title?: string;
  children: React.ReactNode;
}
