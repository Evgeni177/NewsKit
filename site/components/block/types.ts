import React from 'react';
import {
  FlexDirectionProperty,
  JustifyContentProperty,
  AlignItemsProperty,
  DisplayProperty,
  OverflowProperty,
  BorderStyleProperty,
  ColorProperty,
  AlignSelfProperty,
  PositionProperty,
} from 'csstype';
import {ColorKeys, TypePresetKeys, SizingKeys, Responsive} from 'newskit';

export interface BlockProps {
  as?: React.ElementType;
  $color?: Responsive<ColorKeys | ColorProperty>;
  $backgroundColor?: Responsive<ColorKeys | ColorProperty>;
  $font?: Responsive<TypePresetKeys>;
  $border?: Responsive<SizingKeys | string>;
  $borderTop?: Responsive<SizingKeys | string>;
  $borderRight?: Responsive<SizingKeys | string>;
  $borderBottom?: Responsive<SizingKeys | string>;
  $borderLeft?: Responsive<SizingKeys | string>;
  $borderStyle?: Responsive<BorderStyleProperty>;
  $borderWidth?: Responsive<SizingKeys | string>;
  $borderColor?: Responsive<ColorKeys | ColorProperty>;
  $margin?: Responsive<SizingKeys | string>;
  $marginTop?: Responsive<SizingKeys | string>;
  $marginRight?: Responsive<SizingKeys | string>;
  $marginBottom?: Responsive<SizingKeys | string>;
  $marginLeft?: Responsive<SizingKeys | string>;
  $padding?: Responsive<SizingKeys | string>;
  $paddingTop?: Responsive<SizingKeys | string>;
  $paddingRight?: Responsive<SizingKeys | string>;
  $paddingBottom?: Responsive<SizingKeys | string>;
  $paddingLeft?: Responsive<SizingKeys | string>;
  $width?: Responsive<SizingKeys | string>;
  $height?: Responsive<SizingKeys | string>;
  $maxWidth?: Responsive<SizingKeys | string>;
  $maxHeight?: Responsive<SizingKeys | string>;
  $minWidth?: Responsive<SizingKeys | string>;
  $minHeight?: Responsive<SizingKeys | string>;
  $display?: Responsive<DisplayProperty>;
  $flexWrap?: Responsive<boolean>;
  $flexDirection?: Responsive<FlexDirectionProperty>;
  $flexGrow?: Responsive<number>;
  $justifyContent?: Responsive<JustifyContentProperty>;
  $alignItems?: Responsive<AlignItemsProperty>;
  $alignSelf?: Responsive<AlignSelfProperty>;
  $position?: Responsive<PositionProperty>;
  $top?: Responsive<SizingKeys | string>;
  $right?: Responsive<SizingKeys | string>;
  $bottom?: Responsive<SizingKeys | string>;
  $left?: Responsive<SizingKeys | string>;
  $overflow?: OverflowProperty;
}
