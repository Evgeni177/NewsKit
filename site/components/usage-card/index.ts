export * from './usage-card';
export * from './types';
