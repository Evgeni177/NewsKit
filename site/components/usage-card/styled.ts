import {Block, getSpacingFromTheme, styled} from 'newskit';

export const RelativeBlock = styled(Block)`
  position: relative;
`;

export const AbsoluteBlock = styled(Block)`
  position: absolute;
  right: ${getSpacingFromTheme('space040')};
  top: ${getSpacingFromTheme('space040')};
  z-index: 1;
`;
