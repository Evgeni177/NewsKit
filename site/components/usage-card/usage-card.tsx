import React from 'react';
import {Block, Card, TextBlock, Headline, Divider} from 'newskit';
import {CheckIcon} from '../icons/check-icon';
import {CrossIcon} from '../icons/cross-icon';
import {KindConfig, UsageCardProps, UsageKind} from './types';
import {RelativeBlock, AbsoluteBlock} from './styled';

const kindMap: Record<string, KindConfig> = {
  do: {
    heading: 'Do',
    iconComponent: CheckIcon,
    dividerStylePreset: 'dividerPositive',
    headingStylePreset: 'inkPositive',
  },

  dont: {
    heading: 'Do not',
    iconComponent: CrossIcon,
    dividerStylePreset: 'dividerNegative',
    headingStylePreset: 'inkNegative',
  },
};

export const UsageCard: React.FC<UsageCardProps> = ({
  kind = UsageKind.DO,
  description,
  media,
}) => {
  const {
    iconComponent: Icon,
    heading,
    dividerStylePreset,
    headingStylePreset,
  } = kindMap[kind];
  return (
    <RelativeBlock>
      <AbsoluteBlock>
        <Icon />
      </AbsoluteBlock>

      <Card
        media={media}
        overrides={{
          stylePreset: 'cardMediaNonInteractive',
          mediaContainer: {
            stylePreset: 'cardMediaNonInteractive',
            spaceInline: 'space040',
          },
        }}
      >
        <Block spaceStack="space040">
          <Divider overrides={{stylePreset: dividerStylePreset}} />
          <Block spaceStack="space040" />

          <Headline
            overrides={{
              typographyPreset: 'editorialHeadline030',
              heading: {
                stylePreset: headingStylePreset,
              },
            }}
          >
            {heading}
          </Headline>
        </Block>
        <Block spaceStack="space040">
          <TextBlock
            stylePreset="inkBase"
            typographyPreset="editorialParagraph020"
          >
            {description}
          </TextBlock>
        </Block>
      </Card>
    </RelativeBlock>
  );
};
