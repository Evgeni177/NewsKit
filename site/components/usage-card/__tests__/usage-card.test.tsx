import {renderToFragmentWithTheme} from '../../../utils/test-utils';
import {UsageCard} from '../usage-card';
import {UsageCardProps, UsageKind} from '../types';

describe('UsageCard', () => {
  test('Renders default (Do) UsageCard', () => {
    const props: UsageCardProps = {
      description: 'Description',
    };
    const fragment = renderToFragmentWithTheme(UsageCard, props);
    expect(fragment).toMatchSnapshot();
  });

  test('Renders Dont UsageCard when kind=dont', () => {
    const props: UsageCardProps = {
      description: 'Description',
      kind: UsageKind.DONT,
    };
    const fragment = renderToFragmentWithTheme(UsageCard, props);
    expect(fragment).toMatchSnapshot();
  });
});
