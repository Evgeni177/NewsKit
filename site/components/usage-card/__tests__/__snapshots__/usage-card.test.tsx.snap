// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageCard Renders Dont UsageCard when kind=dont 1`] = `
<DocumentFragment>
  .emotion-17 {
  position: relative;
}

.emotion-6 {
  position: absolute;
  right: 16px;
  top: 16px;
  z-index: 1;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 32px;
  padding: 0;
  cursor: default;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
}

.emotion-4 svg {
  width: 16px;
  height: 16px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    max-width: 100%;
  }
}

.emotion-16 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-15 {
  box-sizing: border-box;
}

.emotion-14 {
  box-sizing: border-box;
  padding: 0;
}

.emotion-14 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

.emotion-8 {
  margin-bottom: 16px;
}

.emotion-7 {
  border-width: 0px;
  margin: 0;
  width: 100%;
}

.emotion-10 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-10::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-10::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-12 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-12 svg {
  fill: #2e2e2e;
}

.emotion-12::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-12::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-0 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #d20600;
  vertical-align: unset;
  display: inline-block;
}

.emotion-0.emotion-0 {
  width: 32px;
  height: 32px;
  fill: #d20600;
}

.emotion-9 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #d20600;
}

.emotion-9 svg {
  fill: #d20600;
}

<div
    class="emotion-17"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-4"
          data-testid="flag"
        >
          <div
            class="emotion-3"
          >
            <div
              class="emotion-2"
            >
              <svg
                aria-hidden="true"
                class="emotion-0 emotion-1"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-16"
    >
      <div
        class="emotion-15"
      >
        <div
          class="emotion-14"
        >
          <div
            class="emotion-8"
          >
            <hr
              aria-hidden="true"
              class="emotion-7"
              data-testid="divider"
            />
            <div
              class="emotion-8"
            />
            <section
              class="emotion-10"
            >
              <h1
                class="nk-headline-heading emotion-9"
              >
                Do not
              </h1>
            </section>
          </div>
          <div
            class="emotion-8"
          >
            <p
              class="emotion-12"
            >
              Description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UsageCard Renders default (Do) UsageCard 1`] = `
<DocumentFragment>
  .emotion-17 {
  position: relative;
}

.emotion-6 {
  position: absolute;
  right: 16px;
  top: 16px;
  z-index: 1;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 32px;
  padding: 0;
  cursor: default;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
}

.emotion-4 svg {
  width: 16px;
  height: 16px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    max-width: 100%;
  }
}

.emotion-0 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #0a782a;
  vertical-align: unset;
  display: inline-block;
}

.emotion-0.emotion-0 {
  width: 32px;
  height: 32px;
  fill: #0a782a;
}

.emotion-16 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-15 {
  box-sizing: border-box;
}

.emotion-14 {
  box-sizing: border-box;
  padding: 0;
}

.emotion-14 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

.emotion-8 {
  margin-bottom: 16px;
}

.emotion-7 {
  border-width: 0px;
  margin: 0;
  width: 100%;
}

.emotion-10 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-10::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-10::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-9 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0a782a;
}

.emotion-9 svg {
  fill: #0a782a;
}

.emotion-12 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-12 svg {
  fill: #2e2e2e;
}

.emotion-12::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-12::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-17"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-4"
          data-testid="flag"
        >
          <div
            class="emotion-3"
          >
            <div
              class="emotion-2"
            >
              <svg
                aria-hidden="true"
                class="emotion-0 emotion-1"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41L9 16.17z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-16"
    >
      <div
        class="emotion-15"
      >
        <div
          class="emotion-14"
        >
          <div
            class="emotion-8"
          >
            <hr
              aria-hidden="true"
              class="emotion-7"
              data-testid="divider"
            />
            <div
              class="emotion-8"
            />
            <section
              class="emotion-10"
            >
              <h1
                class="nk-headline-heading emotion-9"
              >
                Do
              </h1>
            </section>
          </div>
          <div
            class="emotion-8"
          >
            <p
              class="emotion-12"
            >
              Description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
